tests/script_annotations_inputs_regression/script_annotations_artifacts_oss_old.py

```diff
@@ -1,22 +1,26 @@
 """Regression test: compare the new Annotated style inputs declaration with the old version."""
 
-from hera.workflows import Workflow, script
-from hera.workflows.artifact import Artifact, OSSArtifact
-from hera.workflows.steps import Steps
+from hera.workflows import (
+    Artifact,
+    OSSArtifact,
+    Steps,
+    Workflow,
+    models as m,
+    script,
+)
 
 
 @script(
     inputs=[
         OSSArtifact(
             name="my_artifact",
             path="/tmp/file",
-            access_key_secret={"name": "my-oss-credentials", "key": "secretKey"},
+            access_key_secret=m.SecretKeySelector(name="my-oss-credentials", key="secretKey"),
+            secret_key_secret=m.SecretKeySelector(name="my-oss-credentials", key="secretKey"),
             bucket="test-bucket-name",
             create_bucket_if_not_present=True,
             endpoint="http://oss-cn-hangzhou-zmf.aliyuncs.com",
             key="test/mydirectory/",
-            lifecycle_rule={"name": "my-oss-rule", "key": "ruleKey"},
```

Comment:
Is it worth adding an actual lifecycle rule that matches the spec? https://github.com/argoproj-labs/hera/blob/961963ba6c2866006ebd850e3c128d81a596ad17/src/hera/workflows/models/io/argoproj/workflow/v1alpha1.py#L279-L292